src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -354,9 +354,14 @@ public function getListAction(int $productId, ProductCombinationFilters $combina
     public function getCombinationIdsAction(int $productId, ProductCombinationFilters $filters): JsonResponse
     {
         /** @var CombinationRepository $repository */
-        $repository = $this->get('prestashop.adapter.product.combination.repository.combination_repository');
+        $repository = $this->get('PrestaShop\PrestaShop\Adapter\Product\Combination\Repository\CombinationRepository');
```

Comment:
I think this modification is not only related to the refacto, if I remember well this API is mostly used by the edition modal so that it has the list of combinations to edit when you use Previous/Next buttons

So this API should be adapted to shop context, but it can be handled separately in one of your PR for combinations multishop because currently there is probably a bug (the modal will display the combinations for all shops even if we are in a single shop with less combinations, so the combination list and the combination modal are not in sync).

TLDR; this is more a bug fix that should be moved in a dedicated PR or one of the PR related to combinations multishop

Reply:
extracted this part here:
https://github.com/PrestaShop/PrestaShop/pull/30585